text_generation/causal_lm/cpp/greedy_causal_lm.cpp

```diff
@@ -91,8 +91,15 @@ int main(int argc, char* argv[]) try {
     lm.get_tensor("input_ids").set_shape({BATCH_SIZE, 1});
     position_ids.set_shape({BATCH_SIZE, 1});
     TextStreamer text_streamer{std::move(detokenizer)};
-    // There's no way to extract special token values from the detokenizer for now
-    constexpr int64_t SPECIAL_EOS_TOKEN = 2;
+    auto& model = lm.get_model();
+    auto rt_info = model.get_rt_info();
+
+    int64_t SPECIAL_EOS_TOKEN = 2; // Default value if not found
+    if (rt_info.count("eos_token_id") > 0) {
+        SPECIAL_EOS_TOKEN = std::any_cast<int64_t>(rt_info["eos_token_id"]);
+    } else {
+        throw std::runtime_error("EOS token ID not found in model's runtime information.");
```

Comment:
why are you throwing error and at the same time you are setting the default value?
Since currently openvino_tokenizer always includes this info to the IR let's remove default value and leave error throwing